pkg/deploy/controller/deployment/controller.go

```diff
@@ -197,37 +196,38 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		}
 
 	case deployapi.DeploymentStatusComplete:
-		// Check for test deployment and ensure the deployment scale matches
-		if config, err := deployutil.DecodeDeploymentConfig(deployment, c.codec); err == nil && config.Spec.Test {
-			deploymentScaled = deployment.Spec.Replicas != 0
-			deployment.Spec.Replicas = 0
-		}
-
 		if err := c.cleanupDeployerPods(deployment); err != nil {
 			return err
 		}
 	}
 
-	if deployutil.CanTransitionPhase(currentStatus, nextStatus) || deploymentScaled {
-		deployment.Annotations[deployapi.DeploymentStatusAnnotation] = string(nextStatus)
+	// Update only if we need to transition to a new phase.
+	if deployutil.CanTransitionPhase(currentStatus, nextStatus) {
+		copied, err := deployutil.DeploymentDeepCopy(deployment)
```

Comment:
I think I'd prefer 

```
deployment, err := deployutil.DeploymentDeepCopy(deployment)
```

That eliminates the possibility of anyone accidentally using the wrong "deployment" variable in the rest of the method.